paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/lookup/PrimaryKeyPartialLookupTable.java

```diff
@@ -241,12 +264,14 @@ public void refresh() {
                 }
 
                 for (Split split : splits) {
-                    BinaryRow partition = ((DataSplit) split).partition();
-                    int bucket = ((DataSplit) split).bucket();
-                    List<DataFileMeta> before = ((DataSplit) split).beforeFiles();
-                    List<DataFileMeta> after = ((DataSplit) split).dataFiles();
+                    DataSplit dataSplit = (DataSplit) split;
+                    BinaryRow partition = dataSplit.partition();
+                    int bucket = dataSplit.bucket();
+                    List<DataFileMeta> before = dataSplit.beforeFiles();
+                    List<DataFileMeta> after = dataSplit.dataFiles();
 
                     tableQuery.refreshFiles(partition, bucket, before, after);
+                    totalBuckets.put(partition, dataSplit.totalBuckets());
```

Comment:
`totalBuckets` might be empty because the table is created by older versions of Paimon. In this case, use `bucket` from `table` instead. Make sure that `bucket > 0`. Add a test case for this.